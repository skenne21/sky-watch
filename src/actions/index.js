export const addCompanyBio = (bio) => ({
  type: 'ADD_COMPANY_BIO',
  bio
})

export const companyError = (message) => {
  type: 'ADD_ERROR_BIO',
  message
}