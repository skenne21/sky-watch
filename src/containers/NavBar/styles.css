.NavBar {
  position: absolute;
  top: 15px;
  right: 5px;
  width: 12%;
  height: 80px;
  margin: 0;
  font-size: 1.2em;
  letter-spacing: 1px;
}

.signin {
  display: block;
  color: #FFF;
  text-decoration: none;
  margin: 10px;
}

.createAccount {
  text-decoration: none;
  color: #fff;
  display: block;
  padding-top: 8px;
}

.NavBar button {
  color: #fff;
  background-color: transparent;
  border: none;
  font-size: 1.2em;
  letter-spacing: 2px;
}

.home {
  letter-spacing: 2px;
  text-decoration: none;
  color: #fff;
  display: block;
  padding-top: 8px;
}

@media screen and (max-width: 1000px) {
  .NavBar {
    font-size: 1em;
  }
}

@media screen and (max-width: 800px) {
  .NavBar {
    width: 100%;
    text-align: right;
  }
}

@media screen and (max-width: 420px) {
  .NavBar {
    font-size: 0.7em;
    width: 35%;
  }

  
}