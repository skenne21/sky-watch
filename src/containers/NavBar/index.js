import React, { Component } from 'react';
import { NavLink, Link, withRouter } from 'react-router-dom';
import { auth } from '../../firebase';
import { connect } from 'react-redux';
import * as actions from '../../actions';
import './styles.css';

export class NavBar extends Component {

  removeUser = async () => {
    auth.signOut();
    this.props.removeUser(this.props.user);
  }

  toggleLogin = () => {
    const user = Object.keys(this.props.user);
    const { pathname }  = this.props.history.location;
    
    if (user.length > 1) {
      return (
        <div>
          <button onClick ={this.removeUser}>SignOut</button>
          {
            pathname !== '/' ? <NavLink to='/'>HOME</NavLink> : ''
          }
        </div>
      )
    } else {
      return (
        <div>
          <NavLink className='signin' to='/signin'>SIGN IN</NavLink>
          <NavLink className='createAccount' to='/signup'>NEW ACCOUNT</NavLink>
        </div>
      )
    }
  }

  render() {
    
    return (
      <div className="NavBar">
        {this.toggleLogin()}
      </div>
    )
  }
}

export const mapStateToProps = ({ user }) => ({
  user
});

export const mapDispatchToProps = dispatch => ({
  removeUser: user => dispatch(actions.removeUser(user))
});


export default withRouter(connect(mapStateToProps, mapDispatchToProps)(NavBar))