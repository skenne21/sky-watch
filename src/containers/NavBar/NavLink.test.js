import React from 'react';
import { shallow } from 'enzyme';
import * as actions from '../../actions';
import { mapStateToProps, mapDispatchToProps } from './index';
import NavBar from './index';

describe('NavBar', () => {
  let wrapper, removeUser, user;

  beforeEach(() => {
    removeUser = jest.fn();
    user = {}
    wrapper = shallow(
      <NavBar
        removeUser={removeUser}
        user={user}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should toggleLogin',() => {

  });

  describe('mapStateToProps', () => {
    let user, mockedState;

    beforeEach(() => { 
      user = {
        name:'bob',
        email:'bob@gmail.com',
        id:1
      };

      mockedState = {
        user: user
      }
    });

    it('Should map user to state', () => {
      const expected = { user };
      const mapped = mapStateToProps(mockedState);
      expect(mapped).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    let mapped, mockDispatch, user;

    beforeEach(() => {
      user = {
        name:'bob',
        email:'bob@gmail.com',
        id:1
      };
      mockDispatch = jest.fn();
      mapped = mapDispatchToProps(mockDispatch);
    });

    it('Should dispatch removeUser with the right params', () => {
      mapped.removeUser(user);
      const expected = actions.removeUser(user);
      expect(mockDispatch).toBeCalledWith(expected);
    })
  });
})