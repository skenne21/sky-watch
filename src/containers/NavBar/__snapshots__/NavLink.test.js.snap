// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    history={
        Object {
            "location": Object {},
          }
    }
    removeUser={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="signin"
                to="/signin"
        >
                SIGN IN
        </NavLink>
        <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="createAccount"
                to="/signup"
        >
                NEW ACCOUNT
        </NavLink>
</div>,
      "className": "NavBar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="signin"
            to="/signin"
>
            SIGN IN
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="createAccount"
            to="/signup"
>
            NEW ACCOUNT
</NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "SIGN IN",
            "className": "signin",
            "to": "/signin",
          },
          "ref": null,
          "rendered": "SIGN IN",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "NEW ACCOUNT",
            "className": "createAccount",
            "to": "/signup",
          },
          "ref": null,
          "rendered": "NEW ACCOUNT",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="signin"
                    to="/signin"
          >
                    SIGN IN
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="createAccount"
                    to="/signup"
          >
                    NEW ACCOUNT
          </NavLink>
</div>,
        "className": "NavBar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="signin"
              to="/signin"
>
              SIGN IN
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="createAccount"
              to="/signup"
>
              NEW ACCOUNT
</NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "SIGN IN",
              "className": "signin",
              "to": "/signin",
            },
            "ref": null,
            "rendered": "SIGN IN",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "NEW ACCOUNT",
              "className": "createAccount",
              "to": "/signup",
            },
            "ref": null,
            "rendered": "NEW ACCOUNT",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
